src/gpu-compute/schedule_stage.cc

```diff
@@ -570,20 +570,25 @@ ScheduleStage::fillDispatchList()
 
         // iterate waves in schList to pick one for dispatch
         auto schIter = schList.at(j).begin();
-        bool dispatched = false;
+        auto selected_iter = schList.at(j).end();
+
+        //find the earliest ready wave according to seqNum
+        for (auto iter = schList.at(j).begin();
+            iter != schList.at(j).end(); iter++) {
+            if (iter->second == RFREADY && dispatchReady(iter->first)) {
+                if (selected_iter == schList.at(j).end()) {
+                    selected_iter = iter;
+                } else if
```

Comment:
I am not sure if this confirms to the official gem5 style by moving the entire () onto the next line, but it's probably fine.